crates/ruff_workspace/src/configuration.rs

```diff
@@ -786,7 +786,7 @@ impl LintConfiguration {
                     .chain(selection.extend_fixable.iter())
                     .filter(|s| s.specificity() == spec)
                 {
-                    for rule in selector.rules(&preview) {
+                    for rule in selector.all_rules() {
```

Comment:
If we only apply the change to the initial `let mut fixable_set: RuleSet = RuleSelector::All.all_rules().collect()`, and not here, confusing things can happen... E.g. if you used `select = ["RUF017"]`, then `unfixable = ["RUF"]` wouldn't mark `RUF017` as unfixable, which seems confusing.